### DISCUSSION TITLE
Call Custom Android Service from shared project

### DISCUSSION THREAD

**Discussion by imms20** [unoplatform/uno#11827] [Q&A]
Hi,

I have to an aar file, for which i have created a xamarin binding library. i have tested that library with .net maui and xamain.android as well as xamarin .forms solutions. everything works there fine. 

Now i want to bring my test project into the uno platform and call the android specific code from shared instance on an interface. 

I am actually trying to create my interface (IMyService) in shared code and call it from shared code (which i did ). Now i am stuck at the point,   How to resolve the android implementation (MyService)?

Do we have a very simple sample of such implemenrtation where we can do something similar like dependency service in xamarin or something?

**Comment by imms20** [comment]
@jeromelaban any thoughts on this one?

**Reply by jeromelaban** [reply]
I'm not sure to understand what you're explaining. If you have a sample app where you can reproduce what you're trying to achieve?

**Comment by FrozDark** [comment]
If you have something platform specific and using a shared project template you can call that code directly from any place in the shared code
If you have cross platform Class PCL, you can use DI to call a code for platform
